test/built-ins/TypedArray/prototype/with/index-validated-against-current-length.js

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2024 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-%typedarray%.prototype.with
+description: >
+  The index is validated against the current length.
+info: |
+  %TypedArray%.prototype.with ( index, value )
+
+  ...
+  8. Else, let numericValue be ? ToNumber(value).
+  9. If IsValidIntegerIndex(O, 𝔽(actualIndex)) is false, throw a RangeError exception.
+  ...
```

Comment:
Not adding more steps seems to suggest that a RangeError should be thrown, but the description says that the index should be validated against the current length, which at this step in this test should be the resized length, 5. Eventually the result should be created and set with the initial length, 2, as the tests confirms.

```suggestion
  1. Let O be the this value.
  2. Let taRecord be ? ValidateTypedArray(O, SEQ-CST).
  3. Let len be TypedArrayLength(taRecord).
  ...
  8. Else, let numericValue be ? ToNumber(value).
  ...
  10. Let A be ? TypedArrayCreateSameType(O, « 𝔽(len) »).
  ...
  13. Return A.
```